### DISCUSSION TITLE
consent.exe endless loop

### DISCUSSION THREAD

**Discussion by danielmarschall** [rustdesk/rustdesk#10390] [General]
### Bug Description

Today I had a connection to a Win10 virtual machine, and I noticed it became very slow as soon as I connected with RustDesk.

Looking with Process Explorer, I see that there is an endless-loop of "tasklist | findstr consent.exe" is running.

Looking at the source code, video_service.rs seems to be the only place where this consent-check is being called from. https://github.com/rustdesk/rustdesk/blob/master/src/server/video_service.rs#L252 The source code comment says "_Note: This function is extremely expensive, do not call it frequently_." , which is strange because this process is *IS* calling is frequently; it is starting/stopping non-stop on the remote computer.

### How to Reproduce

Start RD Connection on a Win10 Virtual Machine

### Expected Behavior

No CPU Load

### Operating system(s) on local (controlling) side and remote (controlled) side

Win10

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.1

### Screenshots

![image](https://github.com/user-attachments/assets/13e5fe23-5166-42e5-8fea-6353b9f5c935)

**Comment by fufesou** [comment]
Thanks.

The comment is used to explain why we add some checks before calling it. `is_process_consent_running()` is expensive.

But we have to check if UAC prompt dialog is shown in "Privacy mode" (the magnifier implementation) and RustDesk is not installed.

`tasklist | findstr consent.exe` should not be called frequenly in most cases.
Maybe this function is called from somewhere else than https://github.com/rustdesk/rustdesk/blob/master/src/server/video_service.rs#L252

I'll check it later.

**Comment by fufesou** [comment]
I'm not sure if there's a better way to check if UAC prompt dialog is showing.

**Comment by danielmarschall** [comment]
Is there any additional information I can provide or help? Unfortunately, I have only access to the computer if I ask the customer to give me access, so my ability to test things are limited.

I am wondering, why does this endless-loop happen only on this computer and not on other computers? Is it something with the remove videocard of that VM? (I expect videocard problems, because the source file is called video_service.rs?)

**Comment by tmr-crypto** [comment]
Same issue here, even if privacy mode is disabled on both sides. Maybe function could be called only when privacy mode enabled, otherwise it is not needed? Thanks
